### DISCUSSION TITLE
It seems that I'm doing something wrong and "await" doesn't work. Any ideas?

### DISCUSSION THREAD

**Discussion by rempas** [oven-sh/bun#7426] [Q&A] [answered]
Hello, I have spent a couple of hours today to build a script (a single file that I don't want to split). It is quite big but in case someone wants to read it, you can [have it](https://gist.github.com/rempas/62f6927bfed729dcdac838c336ab3c50). It is meant to be placed [here](https://github.com/vnmakarov/mir/tree/master/c-benchmarks) and run from the parent directory.

I run it with: `bun run c-benchmarks/bench.ts` and I have the following errors:

```
193 |         const size = await new Response(proc.stdout)
                                 ^
error: Expected ";" but found "new"
    at /home/rempas/Projects/Programming/MIR/c-benchmarks/bench.ts:193:28

265 |     await proc.exited;
                ^
error: Expected ";" but found "proc"
    at /home/rempas/Projects/Programming/MIR/c-benchmarks/bench.ts:265:11

265 |     await proc.exited;
                    ^
error: Unexpected .
    at /home/rempas/Projects/Programming/MIR/c-benchmarks/bench.ts:265:15
```

So yeah, like the titles says, I think that I have something going one with "await". I run the latest version of bun if that helps.

**Comment by alexeyten** [comment, marked as answer]
You’ve forgot to declare function `compile` as async.

By the way, `new` cannot be asynchronous, meaning that there is no reason to write `await new …`

**Reply by rempas** [reply]
> You’ve forgot to declare function compile as async.

OMG!!!!! How can I forgot! And I think the error message should point that out? Should I open an issue?

> By the way, new cannot be asynchronous, meaning that there is no reason to write await new …

So, I will just drop the `new` right? Or will I add `new` before `await`?
